# Avances
## Grupo 2

##### Importación de librerías.

In [1]:
import pandas as pd

##### Importación de datos

In [2]:
df = pd.read_csv('DATOS_CRUDOS.csv')
df.head()

,CODIGO,DISTRITO,DEPARTAMENTO,MUNICIPIO,ESTABLECIMIENTO,DIRECCION,TELEFONO,SUPERVISOR,DIRECTOR,NIVEL,SECTOR,AREA,STATUS,MODALIDAD,JORNADA,PLAN,DEPARTAMENTAL
0,16-01-0137-46,16-006,ALTA VERAPAZ,COBAN,INSTITUTO MIXTO NOCTURNO FRANCISCO MARROQUIN,6A. AVENIDA 1-15 ZONA 4,NaN,JORGE EDUARDO PAQUE LAZARO,--,DIVERSIFICADO,PRIVADO,URBANA,CERRADA DEFINITIVAMENTE,MONOLINGUE,NOCTURNA,DIARIO(REGULAR),ALTA VERAPAZ
1,16-01-0138-46,16-01-0926,ALTA VERAPAZ,COBAN,COLEGIO COBAN,KM.2 SALIDA A SAN JUAN CHAMELCO ZONA 8,77945104,JOSE ARTURO CHOC CHEN,GUSTAVO ADOLFO SIERRA POP,DIVERSIFICADO,PRIVADO,URBANA,ABIERTA,MONOLINGUE,MATUTINA,DIARIO(REGULAR),ALTA VERAPAZ
2,16-01-0139-46,16-01-0927,ALTA VERAPAZ,COBAN,COLEGIO PARTICULAR MIXTO VERAPAZ,KM 209.5 ENTRADA A LA CIUDAD,58015763,ARMANDO AUDILIO POP CUCUL,GILMA DOLORES GUAY PAZ DE LEAL,DIVERSIFICADO,PRIVADO,URBANA,ABIERTA,MONOLINGUE,MATUTINA,DIARIO(REGULAR),ALTA VERAPAZ
3,16-01-0140-46,16-01-0926,ALTA VERAPAZ,COBAN,"COLEGIO ""LA INMACULADA""",7A. AVENIDA 11-109 ZONA 6,78232301,JOSE ARTURO CHOC CHEN,VIRGINIA SOLANO SERRANO,DIVERSIFICADO,PRIVADO,URBANA,ABIERTA,MONOLINGUE,MATUTINA,DIARIO(REGULAR),ALTA VERAPAZ
4,16-01-0141-46,16-01-0924,ALTA VERAPAZ,COBAN,ESCUELA NACIONAL DE CIENCIAS COMERCIALES,2A CALLE 11-10 ZONA 2,40389968,MOISÉS ADRIÁN LÓPEZ PÉREZ,ESTELA DEL CARMEN QUIM ROSALES,DIVERSIFICADO,OFICIAL,URBANA,ABIERTA,MONOLINGUE,MATUTINA,DIARIO(REGULAR),ALTA VERAPAZ


# DIAGNÓSTICO
### A. Número de registros y variables.

In [3]:
n_registros, n_variables = df.shape
print('La cantidad de registros es: ', n_registros)
print('La cantidad de variables es: ', n_variables)

La cantidad de registros es:  9728
La cantidad de variables es:  17


El conjunto de datos contiene 9,728 registros y 17 variables.
Después de eliminar los 22 registros completamente vacíos, el conjunto queda conformado por 9,706 registros.

### B. Tipo de dato de cada variable.
Todas las variables fueron importadas con tipo de dato object, ya que contienen texto o combinaciones de texto y números.

In [4]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9728 entries, 0 to 9727
Data columns (total 17 columns):
 #   Column           Non-Null Count  Dtype 
---  ------           --------------  ----- 
 0   CODIGO           9706 non-null   object
 1   DISTRITO         9405 non-null   object
 2   DEPARTAMENTO     9706 non-null   object
 3   MUNICIPIO        9706 non-null   object
 4   ESTABLECIMIENTO  9703 non-null   object
 5   DIRECCION        9637 non-null   object
 6   TELEFONO         9024 non-null   object
 7   SUPERVISOR       9402 non-null   object
 8   DIRECTOR         8510 non-null   object
 9   NIVEL            9706 non-null   object
 10  SECTOR           9706 non-null   object
 11  AREA             9706 non-null   object
 12  STATUS           9706 non-null   object
 13  MODALIDAD        9706 non-null   object
 14  JORNADA          9706 non-null   object
 15  PLAN             9706 non-null   object
 16  DEPARTAMENTAL    9706 non-null   object
dtypes: object(17)
memory usage: 1.3+ 

Las 17 variables son categóricas. No fue necesario convertir variables a tipos numéricos, ya que incluso los códigos y teléfonos se manejan como identificadores.

### C. Cantidad y porcentaje de valores faltantes por variable.
Antes de la limpieza se identificaron valores faltantes en varias columnas.

In [5]:
faltantes = {'Cantidad': [],
             'Porcentaje %': []}

for x in df.columns:
    cantidad = int(df[x].isna().sum())
    porcentaje = cantidad*100/n_registros

    faltantes['Cantidad'].append(cantidad)
    faltantes['Porcentaje %'].append(porcentaje)

faltantes_df = pd.DataFrame(faltantes, index= df.columns)
faltantes_df

,Cantidad,Porcentaje %
CODIGO,22,0.226151
DISTRITO,323,3.320312
DEPARTAMENTO,22,0.226151
MUNICIPIO,22,0.226151
ESTABLECIMIENTO,25,0.256990
DIRECCION,91,0.935444
TELEFONO,704,7.236842
SUPERVISOR,326,3.351151
DIRECTOR,1218,12.520559
NIVEL,22,0.226151


#### Registros vacíos

In [6]:
print('Hay', len(df[df.isnull().all(axis=1)]), 'registros vacíos.')

Hay 22 registros vacíos.


In [7]:
df[df.isnull().all(axis=1)]

,CODIGO,DISTRITO,DEPARTAMENTO,MUNICIPIO,ESTABLECIMIENTO,DIRECCION,TELEFONO,SUPERVISOR,DIRECTOR,NIVEL,SECTOR,AREA,STATUS,MODALIDAD,JORNADA,PLAN,DEPARTAMENTAL
475,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
647,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
1083,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
1323,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
1482,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
2191,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
4100,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
4692,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
5106,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
5290,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN


- Los únicos registros con código NaN, son aquellos en los que todos los valores son NaN. Debido a esto, se eliminarán estos registros.

In [8]:
df = df.drop(df[df['CODIGO'].isna()].index)

#### Revisión de que registros quedaron con valores faltantes

In [9]:
faltantes = {'Cantidad': [],
             'Porcentaje %': []}

for x in df.columns:
    cantidad = int(df[x].isna().sum())
    porcentaje = cantidad*100/n_registros

    faltantes['Cantidad'].append(cantidad)
    faltantes['Porcentaje %'].append(porcentaje)

faltantes_df = pd.DataFrame(faltantes, index= df.columns)
faltantes_df[faltantes_df['Cantidad'] > 0]

,Cantidad,Porcentaje %
DISTRITO,301,3.094161
ESTABLECIMIENTO,3,0.030839
DIRECCION,69,0.709293
TELEFONO,682,7.010691
SUPERVISOR,304,3.125000
DIRECTOR,1196,12.294408


- Al verificar nuevamente que columnas tienen valores faltantes, solo se encontraron en las columnas de Distrito, Establecimiento, Dirección, Teléfono, Supervisor y Director. Mientras que las demás columnas no tienen datos faltantes.

### D. Cantidad de valores únicos.

In [10]:
unicos = {'Cantidad': []}

for x in df.columns:
    cantidad = int(df[x].nunique())
    unicos['Cantidad'].append(cantidad)

unicos_df = pd.DataFrame(unicos, index= df.columns)
unicos_df

,Cantidad
CODIGO,9706
DISTRITO,1594
DEPARTAMENTO,22
MUNICIPIO,330
ESTABLECIMIENTO,5036
DIRECCION,6023
TELEFONO,5546
SUPERVISOR,1243
DIRECTOR,4854
NIVEL,1


Estas cantidades son coherentes con la naturaleza de cada variable.

### E. Cantidad de registros duplicados exactos.

In [11]:
print('Hay en total:', df.duplicated().sum(), 'datos duplicados.')

Hay en total: 0 datos duplicados.


### F. Variables con valores fuera de dominio o inconsistentes.

In [12]:
df['DEPARTAMENTO'].unique()

array(['ALTA VERAPAZ', 'BAJA VERAPAZ', 'CHIMALTENANGO', 'CHIQUIMULA',
       'EL PROGRESO', 'ESCUINTLA', 'GUATEMALA', 'HUEHUETENANGO', 'IZABAL',
       'JALAPA', 'JUTIAPA', 'PETEN', 'QUETZALTENANGO', 'QUICHE',
       'RETALHULEU', 'SACATEPEQUEZ', 'SAN MARCOS', 'SANTA ROSA', 'SOLOLA',
       'SUCHITEPEQUEZ', 'TOTONICAPAN', 'ZACAPA'], dtype=object)

In [13]:
df['NIVEL'].unique()

array(['DIVERSIFICADO'], dtype=object)

In [14]:
df['SECTOR'].unique()

array(['PRIVADO', 'OFICIAL', 'MUNICIPAL', 'COOPERATIVA'], dtype=object)

In [15]:
df['AREA'].unique()

array(['URBANA', 'RURAL', 'SIN ESPECIFICAR'], dtype=object)

In [16]:
df['STATUS'].unique()

array(['CERRADA DEFINITIVAMENTE', 'ABIERTA', 'CERRADA TEMPORALMENTE',
       'TEMPORAL NOMBRAMIENTO', 'TEMPORAL TITULOS'], dtype=object)

In [17]:
df['MODALIDAD'].unique()

array(['MONOLINGUE', 'BILINGUE'], dtype=object)

In [18]:
df['JORNADA'].unique()

array(['NOCTURNA', 'MATUTINA', 'VESPERTINA', 'DOBLE', 'SIN JORNADA',
       'INTERMEDIA'], dtype=object)

In [19]:
df['PLAN'].unique()

array(['DIARIO(REGULAR)', 'FIN DE SEMANA', 'A DISTANCIA',
       'SEMIPRESENCIAL', 'VIRTUAL A DISTANCIA',
       'SEMIPRESENCIAL (FIN DE SEMANA)',
       'SEMIPRESENCIAL (UN DÍA A LA SEMANA)', 'SABATINO',
       'SEMIPRESENCIAL (DOS DÍAS A LA SEMANA)', 'DOMINICAL', 'MIXTO',
       'INTERCALADO', 'IRREGULAR'], dtype=object)

In [20]:
df['DEPARTAMENTAL'].unique()

array(['ALTA VERAPAZ', 'BAJA VERAPAZ', 'CHIMALTENANGO', 'CHIQUIMULA',
       'EL PROGRESO', 'ESCUINTLA', 'GUATEMALA NORTE', 'GUATEMALA ORIENTE',
       'GUATEMALA OCCIDENTE', 'GUATEMALA SUR', 'HUEHUETENANGO', 'IZABAL',
       'JALAPA', 'JUTIAPA', 'PETÉN', 'QUETZALTENANGO', 'QUICHÉ',
       'QUICHÉ NORTE', 'RETALHULEU', 'SACATEPÉQUEZ', 'SAN MARCOS',
       'SANTA ROSA', 'SOLOLÁ', 'SUCHITEPÉQUEZ', 'TOTONICAPÁN', 'ZACAPA'],
      dtype=object)

In [21]:
pd.Series([len(str(x)) for x in df['TELEFONO']]).unique()

array([ 3,  8,  7, 17,  6, 26,  4, 11,  9, 16,  5, 15, 18,  2, 13])

- Departamental es una variable que divide el departamento de Guatemala en 4 regiones y a Quiché en 2.
- Hay números de teléfono con muchos dígitos (puede que sean varios números) y hay otros con pocos dígitos (menos de 8), los cuales puede que no estén bien escritos.
Se encontró que:
- Los departamentos corresponden a los departamentos oficiales de Guatemala.
- NIVEL únicamente contiene el valor "DIVERSIFICADO".
- SECTOR únicamente presenta:
    - Oficial
    - Privado
    - Municipal
    - Cooperativa
- AREA contiene:
    - Urbana
    - Rural
    - Sin especificar
- STATUS, MODALIDAD, JORNADA y PLAN presentan categorías válidas.


### G. Variables con formatos inconsistentes.

In [22]:
pd.Series([len(str(x)) for x in df['CODIGO']]).unique()

array([13])

CODIGO mantiene una longitud uniforme.

In [23]:
pd.Series([len(str(x)) for x in df['DISTRITO']]).unique()

array([ 6, 10,  3])

In [24]:
df['DISTRITO'].unique()

array(['16-006', '16-01-0926', '16-01-0927', ..., '19-021', '19-10-1140',
       '19-11-1141'], shape=(1595,), dtype=object)

In [25]:
df['MUNICIPIO'].unique()

array(['COBAN', 'SANTA CRUZ VERAPAZ', 'SAN CRISTOBAL VERAPAZ', 'TACTIC',
       'TAMAHU', 'SAN MIGUEL TUCURU', 'PANZOS', 'SENAHU',
       'SAN PEDRO CARCHA', 'SAN JUAN CHAMELCO', 'LANQUIN',
       'SANTA MARIA CAHABON', 'CHISEC', 'CHAHAL',
       'FRAY BARTOLOME DE LAS CASAS', 'LA TINTA', 'RAXRUHA', 'SALAMA',
       'SAN MIGUEL CHICAJ', 'RABINAL', 'CUBULCO', 'GRANADOS',
       'SANTA CRUZ EL CHOL', 'SAN JERONIMO', 'PURULHA', 'CHIMALTENANGO',
       'SAN JOSE POAQUIL', 'SAN MARTIN JILOTEPEQUE', 'SAN JUAN COMALAPA',
       'SANTA APOLONIA', 'TECPAN GUATEMALA', 'PATZUN',
       'SAN MIGUEL POCHUTA', 'PATZICIA', 'SANTA CRUZ BALANYA',
       'ACATENANGO', 'SAN PEDRO YEPOCAPA', 'SAN ANDRES ITZAPA',
       'PARRAMOS', 'ZARAGOZA', 'EL TEJAR', 'CHIQUIMULA',
       'SAN JOSE LA ARADA', 'SAN JUAN ERMITA', 'JOCOTAN', 'CAMOTAN',
       'OLOPA', 'ESQUIPULAS', 'CONCEPCION LAS MINAS', 'QUEZALTEPEQUE',
       'SAN JACINTO', 'IPALA', 'GUASTATOYA', 'MORAZAN',
       'SAN AGUSTIN ACASAGUASTLAN', 'SAN CRIS

In [26]:
pd.Series([len(str(x)) for x in df['ESTABLECIMIENTO']]).unique()

array([ 44,  13,  32,  23,  40,  55,  33,  47,  48,  22,  45,  30,  25,
        21, 105,  87,  39,  27,  36,  59,  24,  94,  46,  42,  51,   3,
        19,  37,  28,  49,  31,  35,  26,  18,  50,  63,  38,  62,  53,
        54,  34,  29,  16,  80,  65,  15,  17,  66,  78,  76,  52,  58,
        73,  64,  43,  71,  57,  68,  60,  41,  81,  61,  70,  77,  79,
       109,  91,  82,  14,  92,  97,  74, 104, 118,  83,  67, 101,  56,
       112,  75,  86,  84,   4,  95, 103,  90,  72,  88,  12,  20,  93,
        69, 100,  96,  99, 125, 121, 122,  89,  11,  85, 107, 113, 123,
        98, 116,   5, 114,   6, 128])

In [27]:
pd.Series([len(str(x)) for x in df['DIRECTOR']]).unique()

array([ 2, 25, 30, 23, 27, 26, 18, 34, 22, 24, 29, 19, 28, 20, 32, 21, 35,
        3, 31, 17, 38, 16, 36, 33, 15,  8,  1,  7, 14, 13, 37, 55, 41, 43,
       42, 52, 40,  4, 39,  6,  5, 47, 11, 45, 51, 50, 10, 49, 12, 44])

- Los códigos tienen 2 distintas longitudes.
- Los distritos son código pero tienen distintas longitudes también.
- Hay establecimientos con longitudes de 5 o menos, por lo que pueden haber errores o falta de información.
- Hay directores que tienen una cantidad pequeña de caractéres y otros que tienen muchos, se trendrán que evaluar.

### H. Identificación de problemas potenciales de calidad de datos.

- Todas las columnas tienen valores nulos.
- Hay columnas en las que los datos tienen longitudes menores o mayores a las esperadas, pudiendo reflejar falta de información o que se tiene más datos que un solo elemento.
- La columna departamental agrega poca información, ya que solo divide a dos departamentos.


### Plan de limpieza.

##### CODIGO

No hay inconsistencias en los registros de esta columna.

##### DISTRITO

In [ ]:
for i, x in df["DISTRITO"].items():

    if pd.isna(x):
        df.at[i, "DISTRITO"] = "NO_ESPECIFICADO"

Si no se especifica el distrito se sustituye por "NO_ESPECIFICADO".

##### DEPARTAMENTO

In [29]:
df["DEPARTAMENTO"] = (
    df["DEPARTAMENTO"]
      .ffill()
      .bfill()
)

Se han rellenado las casillas con valores NaN de departamento con los valores de las casillas inferiores o superiores. Esto se hizo considerando que la lista está en orden alfabetico respecto a estos departamentos.

##### MUNICIPIO

In [30]:
df["MUNICIPIO"] = (
    df["MUNICIPIO"]
      .ffill()
      .bfill()
)

Dado que los municipios de cada departamento están escritos en orden alfabético, se establece que los valores NaN sean sustituidos con los valores de las casillas superiores o inferiores, de manera similar a como se hizo con los departamentos.

##### ESTABLECIMIENTO

In [31]:
for i, x in df["ESTABLECIMIENTO"].items():

    if pd.isna(x):
        df.at[i, "ESTABLECIMIENTO"] = "NO_ESPECIFICADO"

Si no se especifica el establecimiento se sustituye por "NO_ESPECIFICADO".

##### DIRECCION

In [32]:
for i, x in df["DIRECCION"].items():

    if pd.isna(x):
        df.at[i, "DIRECCION"] = "NO_ESPECIFICADO"

Si no se especifica la dirección se sustituye por "NO_ESPECIFICADO".

##### TELEFONO

In [33]:
for i, x in df["TELEFONO"].items():

    if pd.isna(x):
        df.at[i, "TELEFONO"] = "NUMERO_NULO"

    elif len(str(x)) != 8:
        df.at[i, "TELEFONO"] = "NUMERO_INVALIDO"
    

In [34]:
df['TELEFONO']

0       NUMERO_NULO
1          77945104
2          58015763
3          78232301
4          40389968
           ...     
9722    NUMERO_NULO
9723       48579171
9724       55958103
9725       41853380
9726    NUMERO_NULO
Name: TELEFONO, Length: 9706, dtype: object

Se ha sustituido los valores de las casillas con números telefónicos con cantidad de dígitos distinta de 8 con "NUMERO_INVALIDO". Aquellas casillas con valores NaN, se han sustituido con "NUMERO_NULO".

##### SUPERVISOR

In [35]:
for i, x in df["SUPERVISOR"].items():

    if pd.isna(x):
        df.at[i, "SUPERVISOR"] = "NO_ESPECIFICADO"

Si no se especifica el supervisor se sustituye por "NO_ESPECIFICADO".

##### DIRECTOR

In [36]:
for i, x in df["DIRECTOR"].items():

    if pd.isna(x):
        df.at[i, "DIRECTOR"] = "NO_ESPECIFICADO"

Si no se especifica el director se sustituye por "NO_ESPECIFICADO".

##### NIVEL

In [37]:
for i, x in df["NIVEL"].items():

    if pd.isna(x):
        df.at[i, "NIVEL"] = "DIVERSIFICADO"

Se han reemplazado las casillas con valores NaN con "DIVERSIFICADO", ya que el conjunto de datos extraido corresponde a aquel con instituciones de nivel máximo de diversificado.

##### SECTOR

No hay inconsistencias en los registros de esta columna.


##### AREA

No hay inconsistencias en los registros de esta columna.


##### STATUS

No hay inconsistencias en los registros de esta columna.


##### MODALIDAD

No hay inconsistencias en los registros de esta columna.


##### JORNADA

No hay inconsistencias en los registros de esta columna.


##### PLAN

No hay inconsistencias en los registros de esta columna.


##### DEPARTAMENTAL

In [38]:
df=df.drop("DEPARTAMENTAL", axis=1)
df

,CODIGO,DISTRITO,DEPARTAMENTO,MUNICIPIO,ESTABLECIMIENTO,DIRECCION,TELEFONO,SUPERVISOR,DIRECTOR,NIVEL,SECTOR,AREA,STATUS,MODALIDAD,JORNADA,PLAN
0,16-01-0137-46,16-006,ALTA VERAPAZ,COBAN,INSTITUTO MIXTO NOCTURNO FRANCISCO MARROQUIN,6A. AVENIDA 1-15 ZONA 4,NUMERO_NULO,JORGE EDUARDO PAQUE LAZARO,--,DIVERSIFICADO,PRIVADO,URBANA,CERRADA DEFINITIVAMENTE,MONOLINGUE,NOCTURNA,DIARIO(REGULAR)
1,16-01-0138-46,16-01-0926,ALTA VERAPAZ,COBAN,COLEGIO COBAN,KM.2 SALIDA A SAN JUAN CHAMELCO ZONA 8,77945104,JOSE ARTURO CHOC CHEN,GUSTAVO ADOLFO SIERRA POP,DIVERSIFICADO,PRIVADO,URBANA,ABIERTA,MONOLINGUE,MATUTINA,DIARIO(REGULAR)
2,16-01-0139-46,16-01-0927,ALTA VERAPAZ,COBAN,COLEGIO PARTICULAR MIXTO VERAPAZ,KM 209.5 ENTRADA A LA CIUDAD,58015763,ARMANDO AUDILIO POP CUCUL,GILMA DOLORES GUAY PAZ DE LEAL,DIVERSIFICADO,PRIVADO,URBANA,ABIERTA,MONOLINGUE,MATUTINA,DIARIO(REGULAR)
3,16-01-0140-46,16-01-0926,ALTA VERAPAZ,COBAN,"COLEGIO ""LA INMACULADA""",7A. AVENIDA 11-109 ZONA 6,78232301,JOSE ARTURO CHOC CHEN,VIRGINIA SOLANO SERRANO,DIVERSIFICADO,PRIVADO,URBANA,ABIERTA,MONOLINGUE,MATUTINA,DIARIO(REGULAR)
4,16-01-0141-46,16-01-0924,ALTA VERAPAZ,COBAN,ESCUELA NACIONAL DE CIENCIAS COMERCIALES,2A CALLE 11-10 ZONA 2,40389968,MOISÉS ADRIÁN LÓPEZ PÉREZ,ESTELA DEL CARMEN QUIM ROSALES,DIVERSIFICADO,OFICIAL,URBANA,ABIERTA,MONOLINGUE,MATUTINA,DIARIO(REGULAR)
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
9722,19-10-0008-46,19-015,ZACAPA,HUITE,INSTITUTO NACIONAL DE EDUCACION DIVERSIFICADA,NO_ESPECIFICADO,NUMERO_NULO,SILDY MARIELA PEREZ FRANCO,--,DIVERSIFICADO,OFICIAL,RURAL,CERRADA DEFINITIVAMENTE,MONOLINGUE,VESPERTINA,DIARIO(REGULAR)
9723,19-10-0013-46,19-10-1140,ZACAPA,HUITE,INSTITUTO DIVERSIFICADO,BARRIO BUENOS AIRES,48579171,SILDY MARIELA PÉREZ FRANCO,WUENDY JHOJANA SIERRA PAZ,DIVERSIFICADO,OFICIAL,URBANA,ABIERTA,MONOLINGUE,NOCTURNA,DIARIO(REGULAR)
9724,19-10-1009-46,19-10-1140,ZACAPA,HUITE,INSTITUTO DIVERSIFICADO POR COOPERATIVA,BARRIO EL CAMPO,55958103,SILDY MARIELA PÉREZ FRANCO,ROBIDIO PORTILLO SALGUERO,DIVERSIFICADO,COOPERATIVA,URBANA,ABIERTA,MONOLINGUE,VESPERTINA,DIARIO(REGULAR)
9725,19-11-0018-46,19-11-1141,ZACAPA,SAN JORGE,INSTITUTO MIXTO DE EDUCACIÓN DIVERSIFICADA POR...,BARRIO EL CENTRO,41853380,ALBA LUZ MÉNDEZ,VICTOR HUGO GUERRA MONROY,DIVERSIFICADO,COOPERATIVA,URBANA,ABIERTA,MONOLINGUE,MATUTINA,DIARIO(REGULAR)


Dada la poca información dada por esta columna, que es equivalente a la de la columna DEPARTAMENTO, se procede a la eliminación de esta del conjunto.

**Conclusión:**
La limpieza de datos permitió eliminar registros vacíos, tratar valores faltantes e identificar datos inválidos, obteniendo un conjunto de datos más consistente y confiable para su análisis, sin alterar la información relevante de los establecimientos.